src/main/java/StringManipulation.java

```diff
@@ -7,27 +8,120 @@ public String getString() {
 
     @Override
     public void setString(String string) {
+        this.myString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if(myString == null){
+            return 0;
+        }
+
+        int countWords = 0;
+        if(myString.equals("")){
+            return countWords;
+        }
+
+        String removeSpace = myString.trim();
```

Comment:
It looks like you could potentially check if removeSpace.equals("") could be checked before going into the for loop since .trim() should remove all the leading and trailing spaces. In the case of the string "_ _ _ _" (spacing between _'s used to clarify example), spaces would be removed, thus leaving you a string "" and you can return before the loop.